x-pack/metricbeat/module/aws/_meta/docs.asciidoc

```diff
@@ -0,0 +1,54 @@
+This module periodically fetches monitoring metrics from AWS Cloudwatch using GetMetricData API. Note: extra AWS charges on
+GetMetricData API requests will be generated by this module.
+
+The default metricset is `ec2`.
+
+[float]
+=== Module-specific configuration notes
+
+This module environment variable `AWS_ACCESS_KEY_ID`, `AWS_SECRET_ACCESS_KEY`, `AWS_SESSION_TOKEN` and `AWS_REGION`
+references in the config file to set values that need to be configurable during deployment.
+
+There are two different kinds of AWS credentials can be used here: `access keys` and `temporary security credentials`.
+`AWS_ACCESS_KEY_ID` and `AWS_SECRET_ACCESS_KEY` are the two parts of `access keys` for AWS to authenticate AWS API requests.
+`access keys` are long-term credentials for an IAM user or the AWS account root user. Please see
+https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html#access-keys-and-secret-access-keys[AWS Access Keys
+ and Secret Access Keys] for more details. A more AWS recommended way is to use
+`temporary security credentials` instead of `access keys`. `temporary security credentials` consist of an access key ID,
+a secret access key, and a security token which typically returned from `GetSessionToken`. MFA-enabled IAM users would
+need to submit an MFA code while calling `GetSessionToken`. `aws_default_region` is to set the region for SDK to use. Please
+see https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp.html[Temporary Security Credentials] for more details.
+`sts get-session-token` AWS CLI can be used to generate temporary credentials. For example. with MFA-enabled:
+
+----
+aws> sts get-session-token --serial-number arn:aws:iam::1234:mfa/test@elastic.co --token-code 456789 --duration-seconds 129600
+----
+
+By default, Amazon EC2 sends metric data to CloudWatch every 5 minutes. With this basic monitoring, `period` in aws module
+configuration should be larger or equal than `300s`. If `period` is set to be less than `300s`, the same cloudwatch metrics
```

Comment:
I assume in the default on 300s or a multiple of 300s make sense? So if I set it to 400s I get sometimes the same data twice and some data points I could miss?

Reply:
@ruflin If set it to 400s, some data points will be missed. I added some recommendation in this doc. Thanks!